### DISCUSSION TITLE
Hyperparamter Turning

### DISCUSSION THREAD

**Discussion by withkelvin** [ourownstory/neural_prophet#1244] [Q&A - get help using NeuralProphet] [answered]
I just found out about NeuralProphet few days back and decided to give this model a try. However, I am not sure how to tune the hyperparameters, whether to use GridSearchCV or not. May i know what is the preferred way to tune?

**Comment by Alan-Wolf-VN** [comment, marked as answer]
I'm using optuna for hyperparameter tuning. It is much better than GridSearchCV. You can try tuning some hyperparameters such as trend_reg, seasonality_reg, seasonality_mode,n_lags,... Read the documentation to find out which hyperparameters relevant to your problem.

**Reply by withkelvin** [reply]
I tried using GridSearchCV for NeuralProphet but, it wasn't successful because it required XTrain and XTest in fit(). I will have a look at optuna.

**Reply by judussoari** [reply]
Optuna works, I have used it for my project as well :)

**Reply by nehctuk** [reply]
Would it be possible to provide an example?
